core/src/main/java/org/elasticsearch/search/fetch/subphase/highlight/HighlightBuilder.java

```diff
@@ -522,4 +525,33 @@ public String toString() {
             return name().toLowerCase(Locale.ROOT);
         }
     }
+
+    public static enum BoundaryScannerType implements Writeable {
+        SIMPLE, BREAK_ITERATOR;
+
+        public static BoundaryScannerType readFromStream(StreamInput in) throws IOException {
+            int ordinal = in.readVInt();
+            if (ordinal < 0 || ordinal >= values().length) {
+                throw new IOException("Unknown BoundaryScannerType ordinal [" + ordinal + "]");
+            }
+            return values()[ordinal];
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            out.writeVInt(this.ordinal());
+        }
+
+        public static BoundaryScannerType fromString(String boundaryScannerType) {
+            if (boundaryScannerType.toUpperCase(Locale.ROOT).equals(BREAK_ITERATOR.name())) {
+                return BoundaryScannerType.BREAK_ITERATOR;
+            }
+            return SIMPLE;
```

Comment:
I followed the same approach as in `Order` though I agree with you. Just thought this is how ES prefers to do things. I can simplify the method to use `valueOf(boundaryScannerType.toUpperCase(Locale.ROOT))`